samples/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+    "io/ioutil"
+    "log"
+    "path/filepath"
+    "os"
+    "os/exec"
+    "runtime"
+)
+
+func main() {
```

Comment:
Should this be a main run with `go run` or a test run with `go test`?  The latter would allow us to use the "Fail" functions for failed tests while continuing to run additional tests.

Reply:
I didn't think about the different between `go run` and `go test`. Given what you just mentioned, `go test` works better for our testing here.

Reply:
+1 on `go test` and we can use TestMain https://golang.org/pkg/testing/#hdr-Main
for setup/teardown stuff.